packages/spacecat-shared-data-access/src/service/site-candidates/accessPatterns.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { createSiteCandidate } from '../../models/site-candidate.js';
+import { SiteCandidateDto } from '../../dto/site-candidate.js';
+
+/**
+ * Checks if a site candidate exists in site candidates table using base url
+ * @param {DynamoDbClient} dynamoClient - The DynamoDB client.
+ * @param {DataAccessConfig} config - The data access config.
+ * @param {string} baseURL - base url of the site candidate.
+ * @return {Promise<Readonly<Boolean>>} A promise that resolves to true if site candidate exist
+ */
+export const exists = async (dynamoClient, config, baseURL) => {
+  const dynamoItem = await dynamoClient.getItem(config.tableNameSiteCandidates, { baseURL });
+  return dynamoItem !== null;
+};
+
+/**
+ * Adds a site candidate.
+ *
+ * @param {DynamoDbClient} dynamoClient - The DynamoDB client.
+ * @param {DataAccessConfig} config - The data access config.
+ * @param {object} siteCandidateData - The site candidate data.
+ * @returns {Promise<Readonly<SiteCandidate>>} newly created site candidate
+ */
+export const addSiteCandidate = async (
+  dynamoClient,
+  config,
+  siteCandidateData,
+) => {
+  const siteCandidate = createSiteCandidate(siteCandidateData);
+  const siteCandidateExists = await exists(dynamoClient, config, siteCandidate.getBaseURL());
+
+  if (siteCandidateExists) {
```

Comment:
this could also just ignore it because we consciously expect duplicates to be added. may save an exists check for the caller

Reply:
let's keep the duplicates out of the sites-candidates table mainly because coralogix has a 24h sliding window with 50k limit of values. It's likely that we will receive duplicates, and it's better to have a check

Reply:
i am all for the `exists` check - what i mean is you don't have to throw here, but silently ignore the addition request, because the feature seems to be expressly designed with duplicates in mind.

Reply:
now it ignores instead of throwing an error, and the method name changed to `upsertSiteCandidate`